### PR TITLE
Added new Math functions

### PR DESCRIPTION
**Pull request by naurelius**
I added the `log2, hypot, cbrt, gamma, lgamma, erf, erfc` functions. I wrote some tests but that could be followed up

### CODE REVIEW

**Comment by naurelius** [comment]
@ThakeeNathees I don't know why the test keeps failing. It worked fine on my machine 
`assert(cbrt(27) == 3)`

File: tests/lang/core.pk [hunk 1]
```diff
@@ -84,6 +84,19 @@ assert(round(1.4) == 1)
...
 assert(round(-1.5) == -2)
 
+assert(log2(2) == 1)
+assert(log2(1) == 0)
+assert(round(log2(5)) == 2)
+
+assert(hypot(1,1) == sqrt(2))
+assert(round(hypot(3,5)) == 6)
+
+assert(cbrt(27) == 3)
```
Comment (ThakeeNathees) [inline tests/lang/core.pk]: Use the threshold here as well to resolve the issue.

**Review by ThakeeNathees — CHANGES_REQUESTED** [review]
Hi, @victorakpan-dev because of the binary approximation of numbers the floating point arithmetic might differ in different systems by a tiny fraction (see python docs for a good explanation here https://docs.python.org/3/tutorial/floatingpoint.html) That's why we check if the difference between the result and the expected value is less than a tiny threshold value.

**Review by ThakeeNathees — APPROVED** [review]
LGTM, @victorakpan-dev thanks for help us improve the math module,

(I'm traveling now and got a bit busy. I'll test this locally and merge it by tomorrow)

**Comment by ThakeeNathees** [comment]
Hi @victorakpan-dev It's been a long time and the codebase become unmaintained (I've explained here: https://github.com/ThakeeNathees/pocketlang/issues/168#issuecomment-1083429850) anyways thanks for the implementations.

**Pull request merged**